src/lib/customPropTypes.js

```diff
@@ -122,8 +122,12 @@ export const some = (validators) => {
  */
 export const givenProps = (propsShape, validator) => {
   return (props, propName, componentName, ...rest) => {
-    // only validate if all propsShape validators pass
-    if (!_.every(propsShape, (val, key) => !val(props, key, componentName, ...rest))) return
+    const shouldValidate = _.every(propsShape, (val, key) => {
+      // require propShape validators to pass or prop values to match
+      return _.isFunction(val) ? !val(props, key, componentName, ...rest) : val === props[propName]
+    })
```

Comment:
Support defining `giveProps` by value instead of by PropType function only:

```
control: customPropTypes.givenProps(
  { control: 'input' },
  customPropTypes.disallow(['children'])
),
```